tfrecorder/client.py

```diff
@@ -64,19 +64,66 @@ def _validate_runner(
         'DataflowRunner requires a tfrecorder whl file for remote execution.')
 
 
-# def read_image_directory(dirpath) -> pd.DataFrame:
-#   """Reads image data from a directory into a Pandas DataFrame."""
-#
-#   # TODO(cezequiel): Implement in phase 2.
-#   _ = dirpath
-#   raise NotImplementedError
+def _path_split(filepath: str) -> Tuple[str, str]:
+  """Splits `filepath` into (head, tail) where `tail` is the last pathname.
+
+  Similar to `os.path.split` but supports GCS paths (prefix: gs://).
+  """
+
+  if filepath.startswith(constants.GCS_PREFIX):
+    _, path = filepath.split(constants.GCS_PREFIX)
+    head, tail = os.path.split(path)
+    return constants.GCS_PREFIX + head, tail
+
+  return os.path.split(filepath)
+
+
+def _read_image_directory(image_dir: str) -> pd.DataFrame:
+  """Reads image data from a directory into a Pandas DataFrame.
+
+  Expected directory structure:
+    image_dir/
+      <dataset split>/
+        <label>/
+          <image file>
+
+  Example expected directory structure:
+    image_dir/
+      TRAIN/
+        label0/
+          image_000.jpg
+          image_001.jpg
+          ...
+        label1/
+          image_100.jpg
+          ...
+      VALIDATION/
+        ...
+
+  Output will be based on `schema.image_csv_schema`.
+  """
+
```

Comment:
nit: Handle validating the rest of the schema  (perhaps a todo?). Cases to consider:
1) non image files in the split directories (e.g. .DSStore)
2) unsupported image types in the split directories (gif)

Reply:
Good suggestion. Yes I think it would be good to guard against non-image files. I'll add a todo for now.